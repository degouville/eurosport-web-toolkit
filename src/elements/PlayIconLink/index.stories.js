import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, color } from '@storybook/addon-knobs';
import { PlayIconLink } from '../..';

const iconsStories = storiesOf('Elements|PlayIconLink', module);

iconsStories.add('default', () => (
  <PlayIconLink
    bgColorIcon={color('BgColorIcon', '#4652ff')}
    bgColorText={color('bgColorText', '#3c46dc')}
    href={text('href', 'http://eurosport.fr')}
    target={text('target', '')}
  >
    {text('Link Text', 'Replay ')}
  </PlayIconLink>
));
