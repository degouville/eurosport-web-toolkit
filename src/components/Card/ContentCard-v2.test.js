import React from 'react';
import { shallow, mount } from 'enzyme';
import PlayIconLink from '../../elements/PlayIconLink';

import { CardBig, CardSmall, Image } from './ContentCard-v2';

const cardData = {
  image: 'https://i.eurosport.com/2019/04/21/2570541-53309530-2560-1440.jpg?w=800',
  topic: 'Australian Open',
  link: 'https://www.eurosport.fr',
  title: 'Re-Play of the Day: The craziest moments',
  description: 'Roland-garros',
  labelPlayButton: 'Replay',
  darkness: 50,
};

describe('Content card v2', () => {
  it('renders Card with Play Button', () => {
    expect(shallow(<CardBig {...cardData} />)).toMatchSnapshot();
    expect(shallow(<CardSmall {...cardData} />)).toMatchSnapshot();
  });

  it('renders Card without Play Button', () => {
    expect(
      shallow(<CardBig {...cardData} labelPlayButton={undefined} />)
        .dive()
        .find(PlayIconLink).length
    ).toBe(0);
    expect(
      shallow(<CardSmall {...cardData} labelPlayButton={undefined} />)
        .dive()
        .find(PlayIconLink).length
    ).toBe(0);
  });

  it('should render image', () => {
    const wrapper = mount(<CardBig {...cardData} />);
    expect(wrapper.find(Image).length).toEqual(1);
  });

  it('should render placeholder if no image provided', () => {
    // GIVEN
    const wrapper = shallow(<CardBig {...cardData} image={null} />);

    // WHEN
    const images = wrapper.dive().find(Image);
    const { src: imageSrc } = images.first().props();

    // EXPECT
    expect(images.length).toEqual(1);
    expect(imageSrc).toContain('/eurosport-web-toolkit/');
  });
});
