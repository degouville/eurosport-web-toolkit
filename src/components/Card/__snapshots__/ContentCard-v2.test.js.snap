// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content card v2 renders Card with Play Button 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 387px;
}

.emotion-0 .e129r5mb2,
.emotion-0 .e129r5mb3 {
  height: 50%;
}

.emotion-0 .e129r5mb6 {
  margin: 0 24px;
  color: #afafbb;
}

.emotion-0 .e129r5mb7 {
  color: #f8f8f9;
  margin: 11.5px 24px 0 24px;
  font-family: 'Inter UI',-apple-system,-system-ui,system-ui,BlinkMacSystemFont,'Segoe UI',Helvetica,Arial,sans-serif;
  -webkit-font-smoothing: auto;
  -moz-osx-font-smoothing: auto;
  font-smooth: auto;
  font-size: 16px;
  line-height: 24px;
}

.emotion-0 .e129r5mb8 {
  margin: 9px 24px 0 24px;
  color: #afafbb;
}

.emotion-0 .e129r5mb10 {
  bottom: 24px;
  left: 24px;
}

.emotion-0 .exb52wv3 {
  bottom: 190px;
  right: 0;
}

.emotion-0 .e129r5mb2 {
  top: calc(100% - 200px);
}

.emotion-0 .e129r5mb9 {
  top: 211px;
  bottom: auto;
}

@media (min-width:700px) {
  .emotion-0 {
    height: 378px;
  }

  .emotion-0 .e129r5mb2,
  .emotion-0 .e129r5mb3 {
    height: 86%;
  }

  .emotion-0 .e129r5mb6 {
    margin: 0 42px;
    color: #f8f8f9;
  }

  .emotion-0 .e129r5mb7 {
    font-size: 28px;
    line-height: 36px;
    margin: 14px 40px 0 40px;
  }

  .emotion-0 .e129r5mb8 {
    margin: 16px 42px 0 42px;
    color: #f8f8f9;
    text-transform: uppercase;
  }

  .emotion-0 .exb52wv3 {
    bottom: 71px;
    right: 0;
  }

  .emotion-0 .e129r5mb2 {
    top: calc(100% - 57px);
  }

  .emotion-0 .e129r5mb9 {
    top: auto;
    bottom: 99px;
  }

  .emotion-0 .e129r5mb10 {
    bottom: 28px;
    left: 42px;
  }
}

<Component
  className="emotion-0 emotion-1"
  darkness={50}
  description="Roland-garros"
  image="https://i.eurosport.com/2019/04/21/2570541-53309530-2560-1440.jpg?w=800"
  labelPlayButton="Replay"
  link="https://www.eurosport.fr"
  title="Re-Play of the Day: The craziest moments"
  topic="Australian Open"
/>
`;

exports[`Content card v2 renders Card with Play Button 2`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 387px;
}

.emotion-0 .e129r5mb2,
.emotion-0 .e129r5mb3 {
  height: 50%;
}

.emotion-0 .e129r5mb6 {
  margin: 0 24px;
  color: #afafbb;
}

.emotion-0 .e129r5mb7 {
  color: #f8f8f9;
  margin: 11.5px 24px 0 24px;
  font-family: 'Inter UI',-apple-system,-system-ui,system-ui,BlinkMacSystemFont,'Segoe UI',Helvetica,Arial,sans-serif;
  -webkit-font-smoothing: auto;
  -moz-osx-font-smoothing: auto;
  font-smooth: auto;
  font-size: 16px;
  line-height: 24px;
}

.emotion-0 .e129r5mb8 {
  margin: 9px 24px 0 24px;
  color: #afafbb;
}

.emotion-0 .e129r5mb10 {
  bottom: 24px;
  left: 24px;
}

.emotion-0 .exb52wv3 {
  bottom: 190px;
  right: 0;
}

.emotion-0 .e129r5mb2 {
  top: calc(100% - 200px);
}

.emotion-0 .e129r5mb9 {
  top: 211px;
  bottom: auto;
}

@media (min-width:700px) {
  .emotion-0 {
    height: 352px;
  }

  .emotion-0 .e129r5mb9 {
    top: 185px;
  }
}

@media (min-width:1024px) {
  .emotion-0 {
    height: 378px;
  }

  .emotion-0 .e129r5mb9 {
    top: 211px;
  }
}

<Component
  className="emotion-0 emotion-1"
  darkness={50}
  description="Roland-garros"
  image="https://i.eurosport.com/2019/04/21/2570541-53309530-2560-1440.jpg?w=800"
  labelPlayButton="Replay"
  link="https://www.eurosport.fr"
  title="Re-Play of the Day: The craziest moments"
  topic="Australian Open"
/>
`;
